drivers/setup/blue/blue.c

```diff
@@ -911,7 +959,22 @@ ScrIoControl(
 
         case IOCTL_CONSOLE_GET_CURSOR_INFO:
         {
-            PCONSOLE_CURSOR_INFO pcci = (PCONSOLE_CURSOR_INFO)Irp->AssociatedIrp.SystemBuffer;
+            PCONSOLE_CURSOR_INFO pcci;
+
+            /* Validate output buffer */
+            if (Irp->AssociatedIrp.SystemBuffer == NULL)
+            {
+                Status = STATUS_INVALID_USER_BUFFER;
+                break;
+            }
+            if (stk->Parameters.DeviceIoControl.OutputBufferLength < sizeof(CONSOLE_CURSOR_INFO))
+            {
+                Status = STATUS_INVALID_PARAMETER;
+                break;
+            }
+
+            pcci = (PCONSOLE_CURSOR_INFO)Irp->AssociatedIrp.SystemBuffer;
+            RtlZeroMemory(pcsbi, sizeof(CONSOLE_CURSOR_INFO));
```

Comment:
You probably meant `pcci` since **pcsbi** is not declared in IOCTL_CONSOLE_GET_CURSOR_INFO case (concerning the build failure)?

Reply:
Yep, copy-paste error :D